server/src/main/java/com/vaadin/ui/NativeSelect.java

```diff
@@ -203,4 +203,25 @@ public void setEmptySelectionCaption(String caption) {
         Objects.nonNull(caption);
         getState().emptySelectionCaption = caption;
     }
+
+    /**
+     * Sets the number of items that are visible. If only one item is visible,
+     * then the box will be displayed as a drop-down list (the default).
+     *
+     * @param visibleItemCount
+     *            the visible item count
+     */
+    public void setVisibleItemCount(int visibleItemCount) {
+        getState().visibleItemCount = visibleItemCount;
```

Comment:
Check for negative values, throw illegal argument and update javadocs accordingly.